snooty/specparser.py

```diff
@@ -109,7 +62,7 @@ class RefRoleType:
 
 #: Spec definition of a role: this can be either a PrimitiveRoleType, or
 #: an object requiring additional configuration.
-RoleType = Union[PrimitiveRoleType, LinkRoleType, RefRoleType]
+RoleType = Union["PrimitiveRoleType", "LinkRoleType", "RefRoleType"]
```

Comment:
I suspect without proof this might be causing the test failure

Reply:
It's only failing on 3.8, which is why I didn't catch it when I ran `make test` locally, which is INTERESTING…

Reply:
what happens if you remove the double quotes? In *theory* our `from __future__ import annotations` incantation at the top of the file should make that work